sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -361,13 +375,19 @@ def set_xml_body(self, data):
     def set_json_body(self, data):
         """Set a JSON-friendly object as the body of the request.
 
+        Sets your content type header to "application/json" if you
+        didn't input any content type headers
+
         :param data: A JSON serializable object
+        :raises: TypeError
```

Comment:
In what circumstances?

Reply:
Raises one when you try to json.dumps a non serializable object, i.e. a bytes string (for non 2.7)